src/mock_builder.ts

```diff
@@ -52,8 +52,17 @@ export class MockBuilder<T> {
     const original = new this.constructor_fn(...this.constructor_args);
 
     // Attach all of the original's properties to the mock
+    console.log("gonna try attach all props from original, to mock");
     this.getAllProperties(original).forEach((property: string) => {
-      const desc = Object.getOwnPropertyDescriptor(original, property);
+      let desc = Object.getOwnPropertyDescriptor(original, property);
+      if (desc === undefined) {
```

Comment:
Is this different from `null`? If not, can't we simplify to `desc == null` to catch `null` as well?

Reply:
the vreturned value can be undefined, but i change the logic anyways, which inturn checks for null too